data_steward/gcloud/gcs/__init__.py

```diff
@@ -0,0 +1,47 @@
+"""
+Interact with Google Cloud Storage (GCS)
+"""
+# Python stl imports
+
+# Project imports
+
+# Third-party imports
+from google.api_core import page_iterator
+from google.cloud.storage.client import Client
+
+
+class StorageClient(Client):
+    """
+    A client that extends GCS functionality
+    See https://googleapis.dev/python/storage/latest/client.html
+    """
+
+    def list_sub_prefixes(self, bucket: str, prefix: str):
+        """
+        List sub folders in folder specified by prefix
+
+        SO link: https://stackoverflow.com/a/59008580
+        :param bucket: GCS bucket name as string
+        :param prefix: path to directory to look into e.g. a/b/c/
+        :return: list of strings of sub-directories e.g. [a/b/c/v1/, a/b/c/v2/]
+        """
+        if not prefix.endswith('/'):
+            prefix += '/'
+
+        extra_params: dict = {
+            'projection': 'noAcl',
+            'prefix': prefix,
+            'delimiter': '/'
+        }
+
+        path: str = f'/b/{bucket}/o'
```

Comment:
@MichaelSchmidt82 why do have the leading `/b` and trailing `/o`?  I'm not sure we have any paths that will conform to that specification.  To be consistent, we should keep this similar to, https://github.com/all-of-us/curation/blob/develop/data_steward/gcs_utils.py#L131.  This will require updating the unit test.

Reply:
This was used in the original test found in develop.  It's copied and pasted over.  I'll updte the code/test and retest.

Reply:
also: https://github.com/all-of-us/curation/blob/b7d0b5123a97a9c18526edba99cd95277ef0ffc9/data_steward/gcs_utils.py#L64 is not referenced in the project (unused function)

Where as: https://github.com/all-of-us/curation/blob/b7d0b5123a97a9c18526edba99cd95277ef0ffc9/data_steward/utils/gcs.py#L7 (which contains the `f'/b/{bucket}/o'`) is referenced twice.

Reply:
@nishanthpp93 Can you weigh in here?  You added the code referenced [here.](https://github.com/all-of-us/curation/blob/b7d0b5123a97a9c18526edba99cd95277ef0ffc9/data_steward/utils/gcs.py#L7)  If the `/b/` and `/o` are important to the intended function, then I will withdraw my comment about removing them from the path string.

Reply:
@lrwb-aou Sure, I believe it is the [JSON API request form](https://cloud.google.com/storage/docs/folders#rest-apis) of the bucket path. This function is slightly lower level so this form is needed here.

Reply:
Are their any more changes or items to discuss on this PR?